src/module_wrap.cc

```diff
@@ -473,9 +483,11 @@ void ModuleWrap::GetError(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(module->GetException());
 }
 
-MaybeLocal<Module> ModuleWrap::ResolveCallback(Local<Context> context,
-                                               Local<String> specifier,
-                                               Local<Module> referrer) {
+MaybeLocal<Module> ModuleWrap::ResolveModuleCallback(
+    Local<Context> context,
+    Local<String> specifier,
+    Local<FixedArray> import_assertions,
```

Comment:
Not critical because import assertions are behind a harmony flag, but are we supposed to throw an error if the assertions contain something that we do not know?

Reply:
/cc @xtuc @MylesBorins

Reply:
I have two cases in mind:

- `import json from "./package.json" assert { type: "json" }` where we support loading JSON but have not implemented the `type: "json"` assertion yet.
- `import json from "./package.json" assert { foo: "bar" }` where we just don't know what a `foo` assertion is.

Reply:
I believe from reviewing the spec and this thread -> https://github.com/tc39/proposal-import-assertions/issues/27 that it is up to the host to decide what to do with unknown assertions.

pinging @littledan to confirm

Reply:
I opened https://github.com/nodejs/node/issues/37474 to discuss this.